### PR TITLE
ci: enable more linters

### LINKED ISSUE #15: Add golangci-lint to project and CI

### PR DESCRIPTION
**Pull request by ccoVeille**
the default ones are not enough

fixes #15

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      - name: unreachable-code
+
+      # Functions or methods with unused parameters can be a symptom of an unfinished refactoring or a bug.
+      - name: unused-parameter
+
+      # report when a variable declaration can be simplified
+      - name: var-declaration
+
+      # warns when initialism, variable or package naming conventions are not followed.
+      - name: var-naming
```
Comment (chocolacula) [inline .golangci.yml:133]: let's add atomic and wait-group checks

**Review by chocolacula — CHANGES_REQUESTED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  misspell:
+    # Correct spellings using locale preferences for US or UK.
+    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
+    # Default ("") is to use a neutral variety of English.
+    locale: US
```
Comment (chocolacula) [inline .golangci.yml:139]: I don't mind but I'm ok with UK spelling too.

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  revive:
+    rules:
+      # these are the default revive rules
+      # you can remove the whole "rules" node if you want
+      # BUT
+      # ! /!\ they all need to be present when you want to add more rules than the default ones
+      # otherwise, you won't have the default rules, but only the ones you define in the "rules" node
+
+      # Blank import should be only in a main or test package, or have a comment justifying it.
+      - name: blank-imports
```
[the remark this answers is not in the document]
Comment (ccoVeille) [reply .golangci.yml:71]: a blank import is

```go

import(
_ "encoding/json"
)
```

The code is imported but not used

[the remark this answers is not in the document]
Comment (ccoVeille) [reply .golangci.yml:22]: https://github.com/mgechev/revive/blob/master/rule/blank_imports.go

[the remark this answers is not in the document]
Comment (chocolacula) [reply .golangci.yml:22]: Thanks for the explanation

**Review by chocolacula — APPROVED** [review]

**Pull request merged**